pulsar-broker/src/test/java/org/apache/pulsar/client/api/SubscriptionMessageDispatchThrottlingTest.java

```diff
@@ -467,16 +467,16 @@ private void testDispatchRate(SubscriptionType subscription,
         long start = System.currentTimeMillis();
         // Asynchronously produce messages
         for (int i = 0; i < numProducedMessages; i++) {
-            producer.send(new byte[expectRate / 10]);
+            producer.sendAsync(new byte[expectRate / 10]);
         }
         latch.await();
         Assert.assertEquals(totalReceived.get(), numProducedMessages, 10);
         long end = System.currentTimeMillis();
         log.info("-- end - start: {} ", end - start);
 
         // first 10 messages, which equals receiverQueueSize, will not wait.
-        Assert.assertTrue((end - start) >= 2500);
-        Assert.assertTrue((end - start) <= 8000);
+        Assert.assertEquals((end - start), numProducedMessages * expectRate / 10,
```

Comment:
Why is this assertion so certain, it may cause the test to become more unstable

Reply:
It make sense.